core-services/chat/src/main/java/org/silverpeas/core/chat/servers/EJabberdServer.java

```diff
@@ -0,0 +1,173 @@
+/**
+ * Adapter to a remote ejabberd service. It implements the mechanisms to access an ejabberd service
+ * in order to register users and to create networking relationships.
+ * @author mmoquillon
+ */
+@DefaultChatServer
+public class EJabberdServer implements ChatServer {
+
+  private SilverLogger logger = SilverLogger.getLogger(this);
+
+  private final String url;
+
+  private final HttpRequester requester;
+
+  /**
+   * Constructs a new instance.
+   */
+  public EJabberdServer() {
+    SettingBundle settings = ChatServer.getChatSettings();
+    this.url = settings.getString("chat.xmpp.restUrl");
+    String key = settings.getString("chat.xmpp.restKey");
+    this.requester = new HttpRequester("Bearer " + key);
+  }
+
+  @Override
+  public void createUser(final User user) throws ChatServerException {
+    SilverLogger.getLogger(this)
+        .info("Register user {0} ({1})", user.getDisplayedName(), user.getId());
+    ChatUser chatUser = ChatUser.fromUser(user);
+    request("register",
+        o -> o.put("user", chatUser.getChatLogin())
+        .put("host", chatUser.getChatDomain())
+        .put("password", chatUser.getChatPassword()));
+    request("set_vcard",
+        o -> o.put("user", chatUser.getChatLogin())
+        .put("host", chatUser.getChatDomain())
+        .put("name", "FN")
+        .put("content", chatUser.getDisplayedName()));
+  }
+
+  @Override
+  public void deleteUser(final User user) throws ChatServerException {
+    SilverLogger.getLogger(this)
+        .info("Unregister user {0} ({1})", user.getDisplayedName(), user.getId());
+    ChatUser chatUser = ChatUser.fromUser(user);
+    request("unregister",
+        o -> o.put("username", chatUser.getChatLogin())
+            .put("hostname", chatUser.getChatDomain()));
+  }
+
+  @Override
+  public void createRelationShip(final User user1, final User user2) throws ChatServerException {
+    SilverLogger.getLogger(this)
+        .info("Add relationships between {0} ({1}) and {2} ({3})", user1.getDisplayedName(),
+            user1.getId(), user2.getDisplayedName(), user2.getId());
+    ChatUser chatUser1 = ChatUser.fromUser(user1);
+    ChatUser chatUser2 = ChatUser.fromUser(user2);
+    request("add_rosteritem",
+        o -> o.put("localuser", chatUser2.getChatLogin())
+            .put("localserver", chatUser2.getChatDomain())
+            .put("user", chatUser1.getChatLogin())
+            .put("server", chatUser1.getChatDomain())
+            .put("nick", chatUser1.getDisplayedName())
+            .put("subs", "both"));
+    request("add_rosteritem",
+        o -> o.put("localuser", chatUser1.getChatLogin())
+        .put("localserver", chatUser1.getChatDomain())
+        .put("user", chatUser2.getChatLogin())
+        .put("server", chatUser2.getChatDomain())
+        .put("nick", chatUser2.getDisplayedName())
+        .put("subs", "both"));
+  }
+
+  @Override
+  public void deleteRelationShip(final User user1, final User user2) throws ChatServerException {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove the declaration of thrown exception 'org.silverpeas.core.chat.ChatServerException' which is a runtime exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)